markers/uncollect.py

```diff
@@ -70,7 +81,7 @@ def uncollectif(item):
             marker.kwargs.get('reason', 'No reason given'))
 
         try:
-            arg_names = inspect.getargspec(marker._arglist[0][0][0]).args
+            arg_names = inspect.getargspec(get_uncollect_function(marker)).args
```

Comment:
local uncollect worked for me, in the first iterations i got errors precisely because for one of the newly introduced examples the old behaviour was incorrect

additionally the new code will keep working on pytest 3.1, which will change mark internals